python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -18,27 +18,26 @@
 from contextlib import contextmanager
 import functools
 
-import Ganga.Utility.logging
+from Ganga.Utility.logging import getLogger
 
 from copy import deepcopy, copy
-import inspect
+from inspect import isclass
 
-import Ganga.GPIDev.Schema as Schema
+from Ganga.GPIDev.Schema import Schema, Item, ComponentItem, SharedItem
 
 from Ganga.Core.exceptions import GangaValueError, GangaException
 
 from Ganga.Utility.Plugin import allPlugins
 
 def _getName(obj):
     """ Return the name of an object based on what we prioritise"""
-    returnable = getattr(obj, '_name', getattr(obj, '__name__', None))
+    returnable = getattr(obj, '_name', getattr(obj, '__name__', getattr(getattr(obj, '__class__', None), '__name__', None)))
```

Comment:
I guess the question it prompts is "which objects?" At the time we're asking for the name of an object we should know what type it is. It's almost always because it's a `GangaObject` in which case `obj._name` is simpler than `_getName(obj)`. Trying to write some 'god function' which can ascertain the name of an arbitrary object is fraught with danger and causes this function to have to require far too much knowledge about the whole codebase.

There should never be a time where we don't know whether a particular variable is a `GangaObject`, a schema `Item` or some built-in type.